docs/docs/learn/2_privacy_mindset/2_client_side_execution.md

```diff
@@ -34,94 +56,108 @@ Zero-knowledge proofs work similarly but with math. They allow you to prove stat
 
-This property is what makes client-side execution practical. Your device does the heavy lifting of running the computation and generating the proof, while the network only needs to do the lightweight verification.
+This is what makes privacy-preserving blockchains practical: verification cost doesn't scale with computational complexity. Your device does the heavy lifting of running the computation and generating the proof, while the network only needs to do the lightweight verification.
 
-## The prover and verifier dance
+## The prover and verifier
 
 ### Understanding the roles
 
 In the world of zero-knowledge proofs, there are always two parties:
 
-**The Prover** (that's you!):
+**The Prover** (that's your PXE!):
 
-- Executes the actual computation
-- Has access to private inputs (your data)
+- Executes the computation
+- Has access to private inputs (your private data)
 - Generates a mathematical proof of correct execution
 - Sends the proof to the network
 
 **The Verifier** (the Aztec network):
 
 - Receives the proof from the prover
-- Checks the proof's validity using public verification keys
+- Checks the proof's validity using public verification keys (don't worry, we are about to explain verification keys)
 - Accepts or rejects the transaction based on the proof
 - Never sees your private data!
 
 ### Proving and verification keys: The magical key pair
 
-You might be thinking, "How does the verifier know what to check?" This is where proving and verification keys come in. Think of them as a special pair of puzzle pieces that fit together perfectly:
+How does the verifier know what to check?" This is where proving and verification keys come in. Think of them as a special pair of puzzle pieces that fit together perfectly:
+
+[TODO] proving keys??? Not witness????? Or ACIR so like bytecode????
 
-- **Proving Key**: A large file that contains all the mathematical constraints of your program. Your PXE uses this to generate proofs.
+- **Proving Key**: A large file that contains all the mathematical constraints (like specific rules your programme inputs must follow to satisfy the requirements) of your program. Your PXE uses this to generate proofs.
 - **Verification Key**: A small file that contains the essential information needed to verify proofs. The network uses this to check your work.
 
-These keys are generated once when a smart contract is deployed, and they mathematically encode what the contract is supposed to do. It's impossible to generate a valid proof that doesn't follow the contract's rules - the math simply won't work out.
+These keys are generated once when a smart contract is deployed, and they mathematically encode what the contract is supposed to do. It's impossible to generate a valid proof that doesn't follow the contract's rules, the math simply won't work out.
 
 ## Who's who in Aztec?
 
 Let's map this to the Aztec ecosystem:
 
-### When you're the prover
+### You're the prover
```

Comment:
You're the prover for private execution, but the network also proves public execution. It would be good to clarify this distinction